src/TrackGenerator.cpp

```diff
@@ -268,9 +273,27 @@ double TrackGenerator::getZCoord() {
 
 
 /**
+ * @brief Return the array used to store the FSR volumes
+ * @return _FSR_volumes the FSR volumes array indexed by FSR ID
+ */
+FP_PRECISION* TrackGenerator::getFSRVolumesBuffer() {
```

Comment:
We don't want to allocate all new FSR volumes each time we call ``getFSRVolumes``. Also sometimes we might not want to recalculate all the FSR volumes, but rather just get the version previously calculated.